Fails to Build Due to Undefined Symbol Errors

I am on Debian Testing (Bookworm), Kernel version 5.16 on x86_64. As far as I can tell, I have dependencies filled. After running make, I get the following output:
```
ca65 --cpu 6502X -g main.s
main.s(2709): Error: Symbol 'LUT_LINE_TABLE_HI' is undefined
main.s(2719): Note: Expanded from here
main.s(2707): Error: Symbol 'LUT_LINE_TABLE_LO' is undefined
main.s(2717): Note: Expanded from here
main.s(2098): Error: Symbol 'LUT_INTERRUPT_RASTER_TABLE' is undefined
main.s(220): Error: Symbol 'BOTTOM_OF_SCREEN_RASTER' is undefined
main.s(324): Note: Expanded from here
main.s(338): Note: Expanded from here
main.s(350): Note: Expanded from here
tables.s(232): Error: Symbol 'LUT_INTERRUPT_ROUTINES' is undefined
main.s(165): Note: Expanded from here
main.s(167): Note: Expanded from here
main.s(169): Note: Expanded from here
main.s(171): Note: Expanded from here
main.s(350): Note: Expanded from here
tables.s(232): Error: Symbol 'LUT_INTERRUPT_ROUTINES_END' is undefined
main.s(165): Note: Expanded from here
main.s(167): Note: Expanded from here
make: *** [Makefile:26: main.o] Error 1
```

Thanks for the information. Yep. I'm reproducing approximately the same in Ubuntu:

```
ca65 --cpu 6502X -g main.s
main.s(2098): Error: Symbol `LUT_INTERRUPT_RASTER_TABLE' is undefined
main.s(220): Error: Symbol `BOTTOM_OF_SCREEN_RASTER' is undefined
tables.s(232): Error: Symbol `LUT_INTERRUPT_ROUTINES' is undefined
tables.s(232): Error: Symbol `LUT_INTERRUPT_ROUTINES_END' is undefined
make: *** [Makefile:26: main.o] Error 1
```

--- a/src/Makefile
+++ b/src/Makefile
@@ -5,7 +5,8 @@ LD65 = ld65
 PYTHON = python3
 # Part of the Vice emulator project.
 CARTCONV = cartconv
-# This is part of the GoatTracker2 project.
+# This is part of the GoatTracker2 project. On my machine, this was in /usr/lib/goattracker, so it 
+# didn't get added to the PATH automatically when goattracker was installed.
 INS2SND2 = ins2snd2
 # This is slightly customized, so I put it in a submodule.
 EXOMIZER = ../exomizer/exomizer2/src/exomizer

--- a/src/Makefile
+++ b/src/Makefile
@@ -2,7 +2,7 @@ all: fickle.crt
 
 CA65 = ca65
 LD65 = ld65
-PYTHON = python
+PYTHON = python3
 # Part of the Vice emulator project.
 CARTCONV = cartconv
 # This is part of the GoatTracker2 project.
@@ -26,10 +26,10 @@ main.o:	$(SOURCE_FILES) $(GEN_SOURCE_FILES) $(SOUND_EFFECTS) ../Music/music.bin
 	$(CA65) --cpu 6502X -g $<
 
 lineTable.s: buildLineTable.py
-	$(PYTHON) buildLineTable.py > lineTable.s
+	$(PYTHON) buildLineTable.py -o lineTable.s
 
 interruptTables.s: buildInterruptTables.py
-	$(PYTHON) buildInterruptTables.py > interruptTables.s
+	$(PYTHON) buildInterruptTables.py -o interruptTables.s
 
 sfx_%.bin: ../Music/sfx_%.ins
 	$(INS2SND2) $< $@ -b

--- a/src/buildInterruptTables.py
+++ b/src/buildInterruptTables.py
@@ -1,3 +1,14 @@
+from sys import stdout
+from argparse import ArgumentParser, FileType
+
+parser = ArgumentParser()
+parser.add_argument("-o", "--output", action="store")
+parsedArgs = parser.parse_args()
+
+output = stdout
+if parsedArgs.output:
+    output = open(parsedArgs.output, "w")
+
 # Built the table of interrupts and a look-up table based on their raster positions.
 
 first_interrupt = [ 40, "reenableSpritesInterrupt" ]
@@ -23,8 +34,7 @@ interrupt_rows = [
 last_interrupt = [ 50 + (25 * 8), "interrupt" ]
 
 def printInterruptEntry(e):
-	print "\t.byte", e[0], ", .LOBYTE(", e[1] , ")"
-	
+	output.write("\t.byte {}, .LOBYTE({})\n".format(e[0], e[1]))
 
 interrupt_table = [ first_interrupt ]
 current_raster = 50
@@ -34,32 +44,30 @@ for r in interrupt_rows:
 			interrupt_table.append( [ current_raster + column_offsets[i], r[i] ] )
 	current_raster += 21
 
-print "LUT_INTERRUPT_ROUTINES:"
+output.write("LUT_INTERRUPT_ROUTINES:\n")
 for r in interrupt_table:
 	printInterruptEntry(r)
 
-print "BOTTOM_OF_SCREEN_RASTER:"
+output.write("BOTTOM_OF_SCREEN_RASTER:\n")
 printInterruptEntry(last_interrupt)
 
-print "LUT_INTERRUPT_ROUTINES_END:"
-print
-
+output.write("LUT_INTERRUPT_ROUTINES_END:\n\n")
 
 count = 0
 current_raster = interrupt_table[0][0]
 
-print "LUT_INTERRUPT_RASTER_TABLE_BASE:"
-print "LUT_INTERRUPT_RASTER_TABLE = LUT_INTERRUPT_RASTER_TABLE_BASE - ", current_raster
+output.write("LUT_INTERRUPT_RASTER_TABLE_BASE:\n")
+output.write("LUT_INTERRUPT_RASTER_TABLE = LUT_INTERRUPT_RASTER_TABLE_BASE - {}\n".format(current_raster))
 
 # These calcuations are not quite right, but they work!
 
 for r in interrupt_table[1:]:
-	print "\t.byte ",
+	output.write("\t.byte ")
 	while current_raster < r[0]:
-		print count + 3, 
+		output.write("{}".format(count + 3)) 
 		current_raster += 1
 		if current_raster < r[0]:
-			print ",",
-	print
+			output.write(",")
+	output.write("\n")
 	count += 1
 		

--- a/src/buildLineTable.py
+++ b/src/buildLineTable.py
@@ -1,3 +1,14 @@
+from sys import stdout
+from argparse import ArgumentParser, FileType
+
+parser = ArgumentParser()
+parser.add_argument("-o", "--output", action="store")
+parsedArgs = parser.parse_args()
+
+output = stdout
+if parsedArgs.output:
+    output = open(parsedArgs.output, "w")
+
 # The bottom sprite only needs 10 pixels, so splitting at 11
 # means we don't need an extra sprite and extra interrupts.
 splitAt = 11
@@ -14,10 +25,10 @@ def calcLine(x):
 	return d + m
 
 
-print "LUT_LINE_TABLE_HI:"
+output.write("LUT_LINE_TABLE_HI:\n")
 for i in range(0, maxLine + 1):
-	print "\t.byte .hibyte(HUD_RIGHT_ADDRESS + ", calcLine(i), ")"
+	output.write("\t.byte .hibyte(HUD_RIGHT_ADDRESS + {})\n".format(calcLine(i)))
 
-print "LUT_LINE_TABLE_LO:"
+output.write("LUT_LINE_TABLE_LO:\n")
 for i in range(0, maxLine + 1):
-	print "\t.byte .lobyte(HUD_RIGHT_ADDRESS + ", calcLine(i), ")"
+	output.write("\t.byte .lobyte(HUD_RIGHT_ADDRESS + {})\n".format(calcLine(i)))

